clang/test/CodeGenCXX/lambda-this-info.cpp

```diff
@@ -0,0 +1,48 @@
+// RUN: %clang_cl --target=x86_64-windows-msvc /c /Z7 -o %t.obj -- %s
```

Comment:
Generally clang tests don't test end-to-end. If the code change only affects clang IR generation, the test should only test clang IR generation, not all the way down to PDB, etc.

Reply:
I check the IR through the command line:
```
%clang_cc1 -triple x86_64-windows-msvc -emit-llvm --std=c++17 %s -o -
```
and confirmed that this patch will not be reflected in the IR. So, I think it would be better to be placed in ```llvm/test/DebugInfo/COFF```

Reply:
I think you are missing `-gcodeview` in the command line to emit the debug info. The change should reflect on a metadata. `llvm/test/DebugInfo/COFF` is the place to test the phase when IR being lowered to object file.  `clang/test/CodeGenCXX` is for testing C++ code to IR.

Reply:
got it, review again